src/TsChecker.ts

```diff
@@ -1,50 +1,106 @@
+import * as path from "path";
+import { fork, ChildProcess } from "child_process";
 import { DiagnosticError, LintError } from "./util/Error";
-import IncrementalChecker from "./util/IncrementalChecker";
-
-export type TsCheckerResult = {
-  diagnostics: Array<DiagnosticError>;
-  lints: Array<LintError>;
-  time: number;
-};
+import { TsCheckerResult } from "./util/IncrementalChecker";
+import pDefer = require("p-defer");
 
 export default class TsChecker {
-  private incrementalChecker: IncrementalChecker;
+  private process: ChildProcess | null = null;
+  private memoryLimit: number;
+  private tsconfigPath: string;
+  private tslintPath?: string;
+  private exitListener: () => void;
 
-  constructor(tsconfigPath: string, tslintPath?: string) {
-    this.incrementalChecker = new IncrementalChecker(tsconfigPath, tslintPath);
+  constructor(memoryLimit: number, tsconfigPath: string, tslintPath?: string) {
+    this.memoryLimit = memoryLimit;
+    this.tsconfigPath = tsconfigPath;
+    this.tslintPath = tslintPath;
+    this.exitListener = () => {
+      if (this.process != null) {
+        this.process.kill();
+      }
+    };
   }
 
   /**
-   * Checks type checking and linting
+   * Starts the checker process
    */
-  check(): Promise<TsCheckerResult> {
-    const start = Date.now();
+  start() {
+    if (this.process == null) {
+      // terminate children when main process is going to die
+      process.on("exit", this.exitListener);
 
-    return Promise.resolve().then(() => {
-      const { diagnostics, lints } = this.incrementalChecker.run();
-      return {
-        diagnostics,
-        lints,
-        time: Date.now() - start,
-      };
-    });
+      // start child process
+      this.process = fork(
+        process.env.NODE_ENV === "test" ? require.resolve("ts-node/dist/_bin") : require.resolve("./TsCheckerService"),
```

Comment:
process.env.NODE_ENV === "test" should be replaced with something else, as it would break when used as a library.